src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -559,6 +564,11 @@ bool ReplicatedMergeTreeQueue::removeFailedQuorumPart(const MergeTreePartInfo &
 }
 
 std::pair<int32_t, int32_t> ReplicatedMergeTreeQueue::pullLogsToQueue(zkutil::ZooKeeperPtr zookeeper, Coordination::WatchCallback watch_callback, PullLogsReason reason)
+{
+    return pullLogsToQueue(std::make_shared<ZooKeeperWithFaultInjection>(zookeeper), watch_callback, reason);
```

Comment:
Right, `ReplicatedMergeTreeRestartingThread` will not reload zookeeper if a failure is injected with `ZooKeeperWithFaultInjection`. But my code doesn't rely on that anyway. And all other code calling `ReplicatedMergeTreeQueue::pullLogsToQueue()` will not notice my changes also because if a `ZooKeeperWithFaultInjection` is created like in this line:
```
std::make_shared<ZooKeeperWithFaultInjection>(zookeeper)
```
then it works just as a wrapper around `zookeeper`, I mean it doesn't inject any failures.